navigation/navigation-common/src/main/java/androidx/navigation/NavType.kt

```diff
@@ -749,11 +749,18 @@ public abstract class NavType<T> internal constructor(
          */
         @Suppress("RECEIVER_NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
         public override fun parseValue(value: String): D {
+            var caseInsensitiveMatch: D? = null
             for (constant in type.enumConstants) {
-                if ((constant as Enum<*>).name == value) {
+                val enumConstant = (constant as Enum<*>)
+                if (enumConstant.name == value) {
                     return constant
                 }
+                if (enumConstant.name.equals(value, ignoreCase = true)) {
+                    caseInsensitiveMatch = constant
+                    break
```

Comment:
I am missing why we need to break here instead of returning the match?

Reply:
Good call. We don't need to break here. But it probably shouldn't return straight away either. I would think it should continue looping the remaining enum values for an exact match and return that if found, and only return the `caseInsensitiveMatch` if it doesn't end up finding an exact match. 

If you have a deeplink url of `https://myapp.com/someenumvalue`
and an enum:
```
enum class MyEnum {
    SomeEnumValue
    someenumvalue
}
```
We would not want the case insensitive match on the frst item to be returned when the 2nd item will be an exact match.
Let me know if that seems right, and I'll make the appropriate update.

Reply:
I don't think we need to match to that level. We specifically want the Enums to be case insensitive here, so distinguishing by case after that match may not be reasonable.

I would argue that we do want the first item to be returned to discourage this type of confusing distinction. If we look for the exact match, it would allow the following:

```
enum class MyEnum {
RED
REd
Red
red
(etc..)
}
```

Now we are back in the case where all of these could potentially point to different destinations, which is the problem with this bug in the first place. Maybe provide it and hope no one does this type of thing, but I tend to err on the side that if you can do it someone will. @ianhanniballake  curious on your thoughts?

Reply:
OK. If it should just do a simple case-insensitive match, then I can probably update it to use `firstOrNull()` which IMO will make it more concise/readable as well.

Reply:
Yes, let's use `firstOrNull()` here - the first case insensitive match is the one we want to use.

Reply:
@jbw0033 @ianhanniballake - Updated to do a simple case-insensitive match.